aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
+
+.. code-block:: c#
+  :emphasize-lines: 4-5,8-9
+
+  using System.ComponentModel.DataAnnotations;
+  public class LoginViewModel
+  {
+      [Required]
+      [EmailAddress]
+      public string Email { get; set; }
+
+      [Required]
+      [DataType(DataType.Password)]
+      public string Password { get; set; }
+
+      [Display(Name = "Remember me?")]
+      public bool RememberMe { get; set; }
+  }
+
+A controller action:
+
+.. code-block:: c#
+  :emphasize-lines: 3
+  
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null)
+  {
+      if (ModelState.IsValid)
+      {
+        // work with the model
+      }
+      // If we got this far, something failed, redisplay form
+      return View(model);
+  }
+
+The framework will handle validating request data both on the client and on the server. Validation logic specified on model types is added to the rendered views as unobtrusive annotations and is enforced in the browser via `jQuery Validation <http://jqueryvalidation.org/>`__.
+
+Dependency injection
+^^^^^^^^^^^^^^^^^^^^
+
+ASP.NET Core MVC leverages ASP.NET Core's built-in support for :doc:`dependency injection </fundamentals/dependency-injection>`. :doc:`Controllers </mvc/controllers/dependency-injection>` can request needed services through their constructors, allowing them to follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle/>`_:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Controllers/AccountController.cs
+  :lines: 17-38
+  :emphasize-lines: 10-14
+  :language: c#
+
+Your app can also use :doc:`dependency injection in view files </mvc/views/dependency-injection>`, using the ``@inject`` directive:
+
+.. code-block:: html
+  :emphasize-lines: 1
+
+  @inject SomeService ServiceName
+  <!DOCTYPE html>
+  <html>
+  <head>
+    <title>@ServiceName.GetTitle</title>
+  </head>
+  <body>
+    <h1>@ServiceName.GetTitle</h1>
+  </body>
+  </html>
+
+Filters
+^^^^^^^
+
+:doc:`Filters </mvc/controllers/filters>` help developers encapsulate cross-cutting concerns, like exception handling or authorization. Filters enable running custom pre- and post-processing logic for action methods, and can be configured to run at certain points within the execution pipeline for a given request. Filters can be applied to controllers or actions as attributes, and several filters (such as ``Authorize``) are included in the framework.
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Controllers/AccountController.cs
+  :lines: 17-19
+  :emphasize-lines: 1
+  :language: c#
+
+Areas
+^^^^^
+
+:doc:`Areas </mvc/controllers/areas>` provide a way to separate a large app into related groups of models, views, and controllers. This can reduce the number of each of these types of objects that developers must work with as they develop a particular part of the app.
```

Comment:
Replace with 
:doc:`Areas </mvc/controllers/areas>` provide a way to partition a large ASP.NET Core MVC Web app into smaller functional groupings. An area is effectively an MVC structure inside an application. In an MVC project, logical components like Model, Controller, and View are kept in different folders, and MVC uses naming conventions to create the relationship between these components. For a large app, it may be advantageous to partition the app into separate high level areas of functionality. For instance, an e-commerce app with multiple business units, such as checkout, billing, and search etc. Each of these units have their own logical component views, controllers, and models. In this scenario, you can use Areas to physically partition the business components in the same project.

An area can be defined as smaller functional units in an ASP.NET Core MVC project with its own set of controllers, views, and models.